Show loading screen while fetching sites in Pull an existing site modal

epeicher opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "Fixes" keyword and use "Related to" instead.
-->

## Related issues

- Solves STU-1125
- Related to https://github.com/Automattic/studio/pull/2222
- Related to https://github.com/Automattic/studio/pull/2235

## Proposed Changes

- Show the original modal while fetching the sites with the `Loading…` message

https://github.com/user-attachments/assets/612fbffa-eac1-42e7-b5fe-dcb632342d03



## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

- Apply this branch and run `npm start`
- When opening studio, click on `Add site` and `Pull an existing site`
- Check there is a `Loading…` message, if you cannot see the message, try `CMD+R` to refresh data
- Verify that the test steps on linked PR #2222 still work as expected

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [x] Have you checked for TypeScript, React or other console errors?

epeicher committed:
Show loading screen while fetching sites in Pull an existing site modal

katinthehatsite reviewed (approved):
Thanks for fixing this, this worked as expected for me 👍

ivan-ottinger reviewed (commented):
Thanks for the fix, Roberto! The loading message is displaying as expected and the experience is more clear now. 👍🏼

I have noticed one issue - the scrollbars are displayed on the side like this:

Some local sites already exist:

https://github.com/user-attachments/assets/d69c973d-b0f0-4661-984d-1cf3c1e54a57

No local sites:

https://github.com/user-attachments/assets/7dab5231-41a2-4dc1-805a-82b1541d26b0

epeicher committed:
Do not display the list of sites when loading

epeicher commented:
Thanks @katinthehatsite and @ivan-ottinger for your reviews!

> I have noticed one issue - the scrollbars are displayed on the side like this:

Well spotted! That was an existing issue, but I have fixed it as part of [f5594d4](https://github.com/Automattic/studio/pull/2237/commits/f5594d49352b515a2af6233779aed25bbcb27f16). Now the list of sites is no longer rendered while sites are loading. I have also refactored the component a bit for better readability.

ivan-ottinger commented:
> Well spotted! That was an existing issue, but I have fixed it as part of [f5594d4](https://github.com/Automattic/studio/pull/2237/commits/f5594d49352b515a2af6233779aed25bbcb27f16). Now the list of sites is no longer rendered while sites are loading. I have also refactored the component a bit for better readability.

Thank you for addressing that, Roberto! I can confirm the loading is now working correctly! 👌🏼

sejas reviewed (commented):
Thanks for bringing back the loading state.
I confirm it works great. I also tested it with a user without no sites, where it appears the loading state and then switches to the Purchase CTA.
For regular users, the loading state only takes a fraction to load, which is great.

<img width="1041" height="750" alt="Image" src="https://github.com/user-attachments/assets/bc07b85d-e849-4f43-a5cd-90a66b606486" />
